src/Microsoft.Extensions.Configuration.FileProviderExtensions/FileConfigurationProvider.cs

```diff
@@ -0,0 +1,140 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.IO;
+using Microsoft.Extensions.FileProviders;
+
+namespace Microsoft.Extensions.Configuration
+{
+    public abstract class FileConfigurationProvider : ConfigurationProvider
```

Comment:
> I would assume this was factored this way for a reason...

Mainly because different people worked on it (@kirthik and I added FileExtensions, @rynowak and @lodejard added FileProviderExtensions) and that is precisely why we need to refactor now! :smile:

Anyway, I remember there was a reason Physical file provider are considered a heavier/more constraining dependency to have. Maybe @davidfowl or @lodejard can shed some light on it.